app/validators/form_password_validator.rb

```diff
@@ -20,15 +20,18 @@ module FormPasswordValidator
   def strong_password
     return unless errors.messages.blank? && password_score.score < min_password_score
 
-    errors.add :password, :weak_password, **i18n_variables
+    errors.add :password, :weak_password, **i18n_variables, type: :weak_password
   end
 
   def password_graphemes_length
     return if errors.messages.present?
     if password.grapheme_clusters.length < Devise.password_length.min
-      errors.add(:password, :too_short, count: Devise.password_length.min)
+      errors.add(
+        :password, :too_short, count: Devise.password_length.min,
+                               type: :too_short
+      )
```

Comment:
I would indent all of these to the same spot:

```suggestion
      errors.add(
        :password,
        :too_short,
        count: Devise.password_length.min,
        type: :too_short,
      )
```